server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java

```diff
@@ -129,110 +162,88 @@
-                                }
-                            }
-                            if (!isConversionSuccessful) {
-                                throw new ParseException(rawVal, 0);
-                            }
-                            break;
-                        case DATE:
-                            oneField.val = new Date(dateSDF.get().parse(rawVal).getTime());
+                            converted = (oneField.val != null) ? new String((byte[]) oneField.val) : "null";
                             break;
                         default:
-                            throw new UnsupportedOperationException(
-                                    String.format("Field type '%s' (column '%s') is not supported",
-                                            DataType.get(oneField.type),
-                                            column));
+                            converted = (oneField.val != null) ? oneField.val.toString() : "null";
                     }
+                    LOG.debug("Column {} OneField: type {}, content '{}'", columnIndex, oneFieldType, converted);
                 }
             }
 
-            column_index += 1;
+            // Convert TEXT columns into native data types
+            if ((oneFieldType == DataType.TEXT) && (columnType != DataType.TEXT)) {
+                oneField.type = columnType.getOID();
+
+                if (oneField.val == null) {
+                    continue;
+                }
+
+                String rawVal = (String) oneField.val;
+                switch (columnType) {
+                    case VARCHAR:
+                    case BPCHAR:
+                    case TEXT:
+                    case BYTEA:
+                        break;
+                    case BOOLEAN:
+                        oneField.val = Boolean.parseBoolean(rawVal);
+                        break;
+                    case INTEGER:
+                        oneField.val = Integer.parseInt(rawVal);
+                        break;
+                    case FLOAT8:
+                        oneField.val = Double.parseDouble(rawVal);
+                        break;
+                    case REAL:
+                        oneField.val = Float.parseFloat(rawVal);
+                        break;
+                    case BIGINT:
+                        oneField.val = Long.parseLong(rawVal);
+                        break;
+                    case SMALLINT:
+                        oneField.val = Short.parseShort(rawVal);
+                        break;
+                    case NUMERIC:
+                        oneField.val = new BigDecimal(rawVal);
+                        break;
+                    case TIMESTAMP:
+                        oneField.val = new Timestamp(LocalDateTime.parse(rawVal, TIMESTAMP_FORMATTER).toInstant(ZoneOffset.UTC).toEpochMilli());
+                        break;
+                    case DATE:
+                        oneField.val = new Date(DATE_FORMAT.parse(rawVal).getTime());

```

Comment:
From LocalDate.parse it will parse text such as "2007-12-03"
```suggestion
                        oneField.val = Date.valueOf(LocalDate.parse(rawVal));
```

Reply:
wouldn't just java.sql.Date.valueOf(rawVal) work ? https://docs.oracle.com/javase/8/docs/api/java/sql/Date.html#valueOf-java.lang.String-

Reply:
it looks like it would :)

Reply:
Thanks for this comment. As mentioned [here](https://github.com/greenplum-db/pxf/pull/160#discussion_r284143900), the same applies to `Timestamp`.